src/internet_identity_interface/src/lib.rs

```diff
@@ -173,3 +173,92 @@ pub struct HttpResponse {
 pub struct InternetIdentityInit {
     pub assigned_user_number_range: (UserNumber, UserNumber),
 }
+
+// Archive specific types
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub enum OperationType {
+    #[serde(rename = "register_anchor")]
+    RegisterAnchor {
+        initial_device: DeviceDataWithoutAlias,
+    },
+    #[serde(rename = "add_device")]
+    AddDevice { new_device: DeviceDataWithoutAlias },
+    #[serde(rename = "update_device")]
+    UpdateDevice {
+        updated_device: PublicKey,
+        changed_data: DeviceDataUpdate,
```

Comment:
I'll admit that when reading `changed_data` I'm not sure if it's the data that was changed (i.e. the old data) or the current/new data

Reply:
I changed the field name to `new_values` which should hopefully make it more clear.